src/tests.rs

```diff
@@ -199,6 +206,85 @@ fn prop_send_recv_half_closed() {
     QuickCheck::new().tests(7).quickcheck(prop as fn(_))
 }
 
+/// This test simulates two endpoints of a Yamux connection which may be unable to
+/// write simultaneously but can make progress by reading. If both endpoints
+/// don't read in-between trying to finish their writes, a deadlock occurs.
```

Comment:
```suggestion
/// don't read in-between trying to finish their writes, a deadlock occurs.
//
// Ignored for now as the current implementation is prone to the deadlock tested below.
```

With the above would not have to look at the git history.